lib/kuzzle/kuzzle.js

```diff
@@ -305,6 +298,107 @@ class Kuzzle extends KuzzleEventEmitter {
+    // it means that another node has taken care of doing so...
+    // Therefore, there is no need to do the same import twice.
+    const mutexes = types.reduce((result, type) => {
+      result[type] = new Mutex(`backend:import:${type}`, { timeout: 0 });
+      return result;
+    }, {});
+    // Following the same logic as before, mutexes are locked longer
+    // than necessary in order to rarely execute the same import
+    const lockedMutexes = [];
+
+    this.log.info('[ℹ] Imports in progress: This can take some time.');
+
+    try {
+      if (! _.isEmpty(toImport.userMappings) && await mutexes.userMappings.lock()) {
+        lockedMutexes.push(mutexes.userMappings);
+        await this.internalIndex.updateMapping('users', toImport.userMappings);
+        this.log.info('[✔] User mappings import successful');
+      }
+
+      if (await mutexes.mappings.lock()) {
+        lockedMutexes.push(mutexes.mappings);
+        if (! _.isEmpty(toSupport.mappings) && ! _.isEmpty(toImport.mappings)) {
+          throw kerror.get(
+            'plugin',
+            'runtime',
+            'incompatible',
+            '_support.mappings',
+            'import.mappings');
+        }
+        else if (! _.isEmpty(toSupport.mappings)) {
+          await this.ask(
+            'core:storage:public:mappings:import',
+            toSupport.mappings,
+            { rawMappings: true });
+          this.log.info('[✔] Mappings import successful');
+        }
+        else if (! _.isEmpty(toImport.mappings)) {
+          await this.ask('core:storage:public:mappings:import', toImport.mappings);
+          this.log.info('[✔] Mappings import successful');
+        }
+      }
+
+      if (! _.isEmpty(toSupport.fixtures) && await mutexes.fixtures.lock()) {
+        lockedMutexes.push(mutexes.fixtures);
+        await this.ask('core:storage:public:document:import', toSupport.fixtures);
+        this.log.info('[✔] Fixtures import successful');
+      }
+
+      if (await mutexes.securities.lock()) {
+        lockedMutexes.push(mutexes.securities);
+        const isSecuritiesToImport = ! (
+          _.isEmpty(toImport.profiles)
+          && _.isEmpty(toImport.roles)
+          && _.isEmpty(toImport.users)
+        );
+        const isSecuritiesToSupport = toSupport.securities
+          && ! (
+            _.isEmpty(toSupport.securities.profiles)
+            && _.isEmpty(toSupport.securities.roles)
+            && _.isEmpty(toSupport.securities.users)
+          );
+        if (isSecuritiesToSupport && isSecuritiesToImport) {
+          throw kerror.get(
+            'plugin',
+            'runtime',
+            'incompatible',
+            '_support.securities',
+            'import profiles roles or users');
+        }
+        else if (isSecuritiesToSupport) {
+          await this.ask('core:security:load', toSupport.securities, { force: true });
+          this.log.info('[✔] Securities import successful');
```

Comment:
When will we stop using the "securities" word?

https://capital.com/fr/titres-definition

Reply:
Sure I was foolishly repeating the pattern, `permissions` will be more accurate :wink: